editoast/src/views/pagination.rs

```diff
@@ -61,19 +171,27 @@ pub struct PaginatedResponse<T> {
 pub struct PaginationQueryParam {
     #[serde(default = "default_page")]
     #[param(minimum = 1, default = 1)]
-    pub page: i64,
+    pub page: u64,
     #[param(minimum = 1, default = 25)]
-    pub page_size: Option<i64>,
+    pub page_size: Option<u64>,
 }
 
-const fn default_page() -> i64 {
+const fn default_page() -> u64 {
     1
 }
 
 impl PaginationQueryParam {
+    /// Returns a pre-filled [SelectionSettings] from the pagination settings
+    /// that can then be used to list or count models
+    pub fn as_selection_settings<M: Model + 'static>(&self) -> SelectionSettings<M> {
```

Comment:
I've added the implementation of `From`, but:

* it still uses `from_pagination_settings` as we can't import `PaginationQueryParams` into `modelsv2` (well, we can, but we don't want to :D)
* I've kept the function `into_selection_settings` in order to improve readability in the endpoints and because the compiler cannot type check expressions like:

```rust
    let settings = pagination_params
        .validate(1000)?
        .warn_page_size(100)
        .into()
        .order_by(move || ordering.as_project_ordering());
```